pkg/kubelet/prober/prober.go

```diff
@@ -142,9 +146,16 @@ func (pb *prober) runProbe(ctx context.Context, probeType probeType, p *v1.Probe
 		return pb.exec.Probe(pb.newExecInContainer(ctx, container, containerID, command, timeout))
 
 	case p.HTTPGet != nil:
-		req, err := httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
-		if err != nil {
-			return probe.Unknown, "", err
+		var err error
+		key := probeKey{podUID: pod.UID, containerName: container.Name, probeType: probeType}
+		reqAny, found := pb.httpProbeReqCache.Load(key)
+		req, _ := reqAny.(*http.Request)
+		if !found || (p.HTTPGet.Host == "" && req.URL.Hostname() != status.PodIP) {
```

Comment:
This is not very obvious (at least to me). Can you add a comment explaining what you're doing here and why?

Reply:
I added this logic because I knew from many places that a pod is mutable.

As https://github.com/kubernetes/kubernetes/issues/108281#issuecomment-1058503524 states,  
> I don't think we are really saying that the IP can change underneath a running pod.
I'm considering removing this logic totally. If IP for a running pod won't change, can I assume the http probe request for a running pod won't stale?